### PR TITLE
toplevel-info/mgmt: Update to v2/v3

### PR DESCRIPTION
**Pull request by Drakulix**
Swallows up the foreign-toplevel-list protocol handling as well, to be able to guarantee event-order between the cosmic-toplevel-info protocol and foreign-toplevel-list (e.g. it's `done`-event).

Untested, but this should basically work..

### CODE REVIEW

**Comment by Drakulix** [comment]
Also adds now an overlap-notify implementation.

File: src/wayland/protocols/toplevel_info.rs [hunk 1]
```diff
@@ -287,22 +348,32 @@ where
...
                     {
                         handle.closed();
                     }
                 }
                 false
             }
         });
+        for instance in &self.instances {
+            if instance.version() >= zcosmic_toplevel_info_v1::EVT_DONE_SINCE {
+                instance.done();
```
Comment (ids1024) [inline src/wayland/protocols/toplevel_info.rs:365]: Testing this, I'm seeing a flood of `zcosmic_toplevel_info_v1::done` events.

Presumably it shouldn't send `done` unless at least one toplevel has changed (or been added or removed).

**Comment by ids1024** [comment]
Client implementation for toplevel info version 2: https://github.com/pop-os/cosmic-protocols/pull/35. Seems to be working other than the `done` issue. And otherwise this looks reasonable.

File: src/wayland/protocols/toplevel_info.rs (hunk 1 above)
Comment (Drakulix) [reply src/wayland/protocols/toplevel_info.rs]: Should be fixed now. Global `done` also needed some fixes to be only send after we cleaned up all of the state updates successfully.

**Pull request merged**